crypto/cms/cms_kari.c

```diff
@@ -413,13 +429,15 @@ int cms_RecipientInfo_kari_init(CMS_RecipientInfo *ri,  X509 *recip, EVP_PKEY *r
 static int cms_wrap_init(CMS_KeyAgreeRecipientInfo *kari,
                          const EVP_CIPHER *cipher)
 {
+    const CMS_CTX *cms_ctx = kari->cms_ctx;
     EVP_CIPHER_CTX *ctx = kari->ctx;
-    const EVP_CIPHER *kekcipher;
+    EVP_CIPHER *kekcipher;
+    const char *kekcipher_name;
     int keylen;
     int ret;
 
     /* If a suitable wrap algorithm is already set nothing to do */
-    kekcipher = EVP_CIPHER_CTX_cipher(ctx);
+    kekcipher = (EVP_CIPHER *)EVP_CIPHER_CTX_cipher(ctx);
```

Comment:
I'm sure it has got a reason but was just wondering why is it necessary to cast from const...?

Reply:
I was reusing the variable down further - will add it as a seperate non cost variable instead.